task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -1095,6 +1125,16 @@ def _execute_task(context: Context, ti: RuntimeTaskInstance, log: Logger):
     return result
 
 
+def _render_map_index(context: Context, *, jinja_env: jinja2.Environment | None, log: Logger) -> str | None:
+    """Render named map index if the DAG author defined map_index_template at the task level."""
+    if jinja_env is None or (template := context.get("map_index_template")) is None:
```

Comment:
A) how would jinja env ever be None?
B) don't eagerly create (nor pass in!) the jinja env, delay it until we actually know we need it and create it in this fn